File: medium/2294. Partition Array Such That Maximum Difference Is K/2294. Partition Array Such That Maximum Difference Is K.ts
function partitionArray(nums: number[], k: number): number {
    nums.sort((a, b) => a - b)

    let res = 1, prev = nums[0]
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] - prev > k) {
            prev = nums[i]
            res += 1
        }
    }
    return res
};

File: medium/2294. Partition Array Such That Maximum Difference Is K/2294. Partition Array Such That Maximum Difference Is K.py
from typing import List


class Solution:
    def partitionArray(self, nums: List[int], k: int) -> int:
        nums.sort()

        res, prev = 0, nums[0]

        for i in range(1, len(nums)):
            if nums[i] - prev > k:
                prev = nums[i]
                res += 1

        return res
